api/src/main/java/marquez/db/mappers/JobMapper.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.db.mappers;
+
+import static marquez.db.Columns.stringOrNull;
+import static marquez.db.Columns.stringOrThrow;
+import static marquez.db.Columns.timestampOrThrow;
+import static marquez.db.Columns.urlOrNull;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import marquez.common.Utils;
+import marquez.common.models.DatasetId;
+import marquez.common.models.JobId;
+import marquez.common.models.JobName;
+import marquez.common.models.JobType;
+import marquez.common.models.NamespaceName;
+import marquez.db.Columns;
+import marquez.service.models.Job;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.postgresql.util.PGobject;
+
+@Slf4j
+public final class JobMapper implements RowMapper<Job> {
+  public static final ObjectMapper mapper = Utils.getMapper();
+
+  @Override
+  public Job map(@NonNull ResultSet results, @NonNull StatementContext context)
+      throws SQLException {
+    return new Job(
+        new JobId(
+            NamespaceName.of(stringOrThrow(results, Columns.NAMESPACE_NAME)),
+            JobName.of(stringOrThrow(results, Columns.NAME))),
+        JobType.valueOf(stringOrThrow(results, Columns.TYPE)),
+        JobName.of(stringOrThrow(results, Columns.NAME)),
+        timestampOrThrow(results, Columns.CREATED_AT),
+        timestampOrThrow(results, Columns.UPDATED_AT),
+        getDatasetFromJsonOrNull(results, "current_inputs"),
+        getDatasetFromJsonOrNull(results, "current_outputs"),
+        urlOrNull(results, "current_location"),
+        toContext(results, Columns.CONTEXT),
+        stringOrNull(results, Columns.DESCRIPTION),
+        null // toRun(results, "run")
```

Comment:
We'll want to return the latest run for a job if it exists.

Reply:
This is done late in two parts in the job dao. I tried writing the more complex query to resolve the run in one go but ran into a sql.timestamp->instant serialization issue. It'd be solveable but i wanted to keep it simple here.

Reply:
Sounds good. Maybe add a note on fixing the `sql.timestamp->instant serialization` issue at a later time and/or the current logic?